app/uk/gov/hmrc/tradergoodsprofilesdatastore/connectors/RouterConnector.scala

```diff
@@ -195,12 +195,28 @@ class RouterConnector @Inject() (config: Configuration, httpClient: HttpClientV2
       .execute[HttpResponse]
       .flatMap { response =>
         response.status match {
-          case OK        => Future.successful(true)
-          case NOT_FOUND => Future.successful(false)
-          case _         => Future.failed(UpstreamErrorResponse(response.body, response.status))
+          case OK          => Future.successful(true)
+          case NOT_FOUND   => Future.successful(false)
+          case BAD_REQUEST => handleBadRequest(response)
+          case _           => Future.failed(UpstreamErrorResponse(response.body, response.status))
         }
       }
 
+  private def handleBadRequest(response: HttpResponse): Future[Boolean] = {
+    val json = Json.parse(response.body)
+    (json \ "errors").asOpt[List[JsObject]] match {
+      case Some(errors) =>
+        val errorNumbers = errors.flatMap(err => (err \ "errorNumber").asOpt[Int])
+        if (errorNumbers.exists(num => num == 10 || num == 31)) {
```

Comment:
What do these error codes mean? Might be worth adding them to a named constant too e.g error_name = 10